python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -638,6 +649,56 @@ def _find_stars(self, exposure, background):
         else:
             return result.sourceCat
 
+    def _match_psf_stars(self, psf_stars, stars):
+        """Match calibration stars to psf stars, to identify which were psf
+        candidates, and which were used or reserved during psf measurement.
+
+        Parameters
+        ----------
+        psf_stars : `lsst.afw.table.SourceCatalog`
+            Psf candidate stars that were sent to the psf determiner. Used to
+            populate psf-related flag fields.
+        stars : `lsst.afw.table.SourceCatalog`
+            Stars that will be used for calibration; psf-related fields will
+            be updated in-place.
+        """
```

Comment:
It's probably worth commenting here that this code is adapted from https://github.com/lsst/pipe_tasks/blob/4412426be99252095907e56ef8144b7bfd3df66c/python/lsst/pipe/tasks/calibrate.py#L741